externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetCatalogs.scala

```diff
@@ -33,8 +33,12 @@ class GetCatalogs(spark: SparkSession, session: Session)
   }
 
   override protected def runInternal(): Unit = {
-    iter = Seq(
-      Row(spark.sessionState.catalogManager.currentCatalog.name())
-    ).toList.iterator
+     try {
+       iter = try {
+        spark.sql("SELECT CURRENT_CATALOG()").collect().toList.toIterator
+      } catch {
+        case _: AnalysisException => Seq(Row("spark_catalog")).toIterator
```

Comment:
How about fallback to previous action rather than hardcode?

Reply:
The previous one calling unstable API of spark, we can improve this later(like retrieving the default catalog from conf, but none of these are stable yet)

Reply:
I'm afraid -1 for the current approach. I'm using Spark 3.0.1 with Iceberg catalog, the change will broken our use case.
```
spark.sql.defaultCatalog=spark_catalog
spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions
spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog
spark.sql.catalog.iceberg=org.apache.iceberg.spark.SparkCatalog
spark.sql.catalog.iceberg.type=hadoop
spark.sql.catalog.iceberg.warehouse=hdfs://nameservice1/iceberg-warehouse
```

Reply:
Oh I see, but according to your configuration the catalogname is still spark_catalog?

Reply:
There are 2 catalogs: `spark_catalog` and `iceberg`

Reply:
OK, I just notice that the CURRENT_CATALOG does not get merged into Spark 3.0